hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/OnlineDsAssignmentStrategy.java

```diff
@@ -106,4 +167,49 @@ JpaActionStatus createActionStatus(final JpaAction action, final String actionMe
         return result;
     }
 
+    private DistributionSetAssignmentResult sendDistributionSetAssignedEvent(
+            final DistributionSetAssignmentResult assignmentResult) {
+        final List<Action> filteredActions = assignmentResult.getActions().stream().filter(action -> {
+            final Status actionStatus = action.getStatus();
+            return !hasPendingCancellations(action.getTarget()) && Status.CANCELING != actionStatus
+                    && Status.CANCELED != actionStatus;
+        }).collect(Collectors.toList());
+        final DistributionSet set = assignmentResult.getDistributionSet();
+        sendTargetAssignDistributionSetEvent(set.getTenant(), set.getId(), filteredActions);
+        return assignmentResult;
+    }
+
+    private void sendTargetAssignDistributionSetEvent(final String tenant, final long distributionSetId,
+            final List<Action> actions) {
+        if (CollectionUtils.isEmpty(actions)) {
+            return;
+        }
+
+        afterCommit.afterCommit(() -> eventPublisher.publishEvent(new TargetAssignDistributionSetEvent(tenant,
+                distributionSetId, actions, bus.getId(), actions.get(0).isMaintenanceWindowAvailable())));
+    }
+
+    private boolean hasPendingCancellations(final Target target) {
+        return actionRepository.findByActiveAndTarget(null, target.getControllerId(), true).getContent().stream()
```

Comment:
This call will fetch DS from the DB as well. Isn't there a better query available, maybe even with given status you're looking for?

Reply:
yes, a dedicated DB call would be better... will try that